### PR TITLE
[ServiceBus] receive related methods/constructor raise error if given max_wait_time <= 0

### PR DESCRIPTION
**Pull request by yunhaoling**
This PR addresses the issue: https://github.com/Azure/azure-sdk-for-python/issues/14138

* `ServiceBusReceiver.receive_messages/get_streaming_message_iter` and
 `ServiceBusClient.get_<queue/subscription>_receiver` now raises ValueError if the given `max_wait_time` is less than 0.

### CODE REVIEW

File: sdk/servicebus/azure-servicebus/tests/async_tests/test_queues_async.py [hunk 1]
```diff
@@ -63,7 +63,17 @@ async def test_async_queue_by_queue_client_conn_str_receive_handler_peeklock(sel
             with pytest.raises(ServiceBusConnectionError):
                 await (sb_client.get_queue_receiver(servicebus_queue.name, session_id="test", max_wait_time=5))._open_with_retry()
 
+            with pytest.raises(ValueError):
```
Comment (KieranBrantnerMagee) [inline sdk/servicebus/azure-servicebus/tests/async_tests/test_queues_async.py:66]: I'm debating whether having a comparable check for subscription receiver passes my bar for "worth being obnoxious about".  I'll leave that to your judgement and rubber stamp this regardless; if you decide to put it there as well feel free to ping me again for a quick OK.

**Review by KieranBrantnerMagee — APPROVED** [review]
LGTM, but see the above comment/don't hesitate to give me a shout if needed.

File: sdk/servicebus/azure-servicebus/tests/async_tests/test_queues_async.py (hunk 1 above)
Comment (yunhaoling) [reply sdk/servicebus/azure-servicebus/tests/async_tests/test_queues_async.py:72]: good callout, just added several lines of code in subscription receiver tests.

**Review by KieranBrantnerMagee — APPROVED** [review]

File: sdk/servicebus/azure-servicebus/CHANGELOG.md [hunk 2]
```diff
@@ -36,6 +36,9 @@ now raise more concrete exception other than `MessageSettleFailed` and `ServiceB
 * Exceptions `MessageSendFailed`, `MessageSettleFailed` and `MessageLockExpired`
  now inherit from `azure.servicebus.exceptions.MessageError`.
 * `get_state` in `ServiceBusSession` now returns `bytes` instead of a `string`.
+* `encoding` support is removed from `ServiceBusMessage`
```
Comment (yunhaoling) [inline sdk/servicebus/azure-servicebus/CHANGELOG.md:39]:
```suggestion
```

**Review by KieranBrantnerMagee — APPROVED** [review]

**Pull request merged**